proto-lens-tests/tests/proto3_test.hs

```diff
@@ -98,16 +98,20 @@ main = testMain
       , testCase "string" $ (def :: Strings) @=? (def & string .~ "")
       , testCase "enum" $ (def :: Foo) @=? (def & enum .~ Foo'Enum1)
       ]
-  -- Enums are all pattern aliases
+  -- Enums are sum types, except for aliases
   , testGroup "enum"
       [ testCase "aliases are exported" $ Foo'Enum2 @=? Foo'Enum2a
+      , serializeTo "serializeTo enum"
+          (def & enum .~ Foo'Enum2 :: Foo)
+          "enum: Enum2"
+          "0\ETX"
       , testCase "enum values" $ do
           map toEnum [0, 3, 3] @=? [Foo'Enum1, Foo'Enum2, Foo'Enum2a]
-          ["Foo'Enum1", "Foo'Enum2", "Foo'Enum2", "toEnum 5"]
+          ["Foo'Enum1", "Foo'Enum2", "Foo'Enum2", "Foo'FooEnum'Unrecognized (Foo'FooEnum'UnrecognizedValue 5)"]
               @=? map show [Foo'Enum1, Foo'Enum2, Foo'Enum2a, toEnum 5]
           ["Enum1", "Enum2", "Enum2", "6"]
               @=? map showEnum [Foo'Enum1, Foo'Enum2, Foo'Enum2a, toEnum 6]
-          [Just Foo'Enum1, Just Foo'Enum2, Just Foo'Enum2, Nothing, Nothing]
+          [Just Foo'Enum1, Just Foo'Enum2, Just Foo'Enum2, maybeToEnum 4, maybeToEnum 5]
```

Comment:
Test nit (working around the fact that the constructor isn't provided): Add a new test that checks `fromEnum <$> maybeToEnum 4 == Just 4`